File: api/src/lib/paths.ts
import path from "node:path"
import { fileURLToPath } from "node:url"

const __dirname = fileURLToPath(new URL(".", import.meta.url))

// FOLIO_DATA env var overrides — same convention the Go server will use.
// Default: ../../../testdata/folio relative to api/src/lib/
const dataRoot = process.env.FOLIO_DATA
  ? path.resolve(process.env.FOLIO_DATA)
  : path.resolve(__dirname, "../../../testdata/folio")

export const paths = {
  root: dataRoot,
  features: path.join(dataRoot, "features"),
  wiki: path.join(dataRoot, "wiki"),
  issues: path.join(dataRoot, "issues"),
  sprints: path.join(dataRoot, "sprints"),
  team: path.join(dataRoot, "team.md"),
  config: path.join(dataRoot, "folio.yaml"),
  roadmap: path.join(dataRoot, "roadmap.md"),
  aiSessions: path.join(dataRoot, ".sessions"),
  projectDocs: path.join(dataRoot, "project-docs"),
}

File: commandline/internal/store/paths.go
// Package store provides filesystem-backed data stores for Folio entities.
package store

import (
	"os"
	"path/filepath"
)

// Paths holds resolved filesystem paths for all entity types.
type Paths struct {
	Root        string
	Features    string
	Wiki        string
	Issues      string
	Sprints     string
	Team        string
	Config      string
	Roadmap     string
	AISessions  string
	ProjectDocs string
}

// NewPaths creates a Paths instance from a data root directory.
func NewPaths(root string) *Paths {
	return &Paths{
		Root:        root,
		Features:    filepath.Join(root, "features"),
		Wiki:        filepath.Join(root, "wiki"),
		Issues:      filepath.Join(root, "issues"),
		Sprints:     filepath.Join(root, "sprints"),
		Team:        filepath.Join(root, "team.md"),
		Config:      filepath.Join(root, "folio.yaml"),
		Roadmap:     filepath.Join(root, "roadmap.md"),
		AISessions:  filepath.Join(root, ".sessions"),
		ProjectDocs: filepath.Join(root, "project-docs"),
	}
}

// ResolvePaths resolves the data root from FOLIO_DATA env var or a default.
func ResolvePaths(defaultRoot string) *Paths {
	root := os.Getenv("FOLIO_DATA")
	if root == "" {
		root = defaultRoot
	}
	abs, err := filepath.Abs(root)
	if err != nil {
		abs = root
	}
	return NewPaths(abs)
}
